### ISSUE TITLE
Corrupt video shown in Peer connection -> Constraints page

### ISSUE THREAD

**Issue by cvranjani** [webrtc/samples#718]
1. Browse to https://webrtc.github.io/samples/src/content/peerconnection/constraints/
2. Open the javascript console (alt+cmd+j or ctr+shift+j) and watch for any errors
3. Adjust the Camera constraint sliders to following combination:
   maxWidth 1300, minWidth 1000, maxHeight 800, minHeight 690 
4. Press "Get media"
5. Press "Connect"
6. Observe that Remote video shown is corrupted. Please see attached screenshots.
   ![img_7474](https://cloud.githubusercontent.com/assets/15255941/11827218/6f66a97a-a33f-11e5-9282-7f78433e17d8.JPG)
   ![img_7475](https://cloud.githubusercontent.com/assets/15255941/11827221/746bb384-a33f-11e5-83c1-ea4db0c5c88c.JPG)

Expected result:
Remote video should be clear and should result in 1280x720 from most cameras.

Note: There are no javascript errors. Please see the attached logs.

**Comment by cvranjani** [comment]
I could not attach the logs for some reason. Copy/pasting java script logs here:

"This appears to be Chrome
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
3main.js:262 Not connected yet
adapter.js:32 chrome: {"audio":true,"video":{"mandatory":{"minWidth":"300","maxWidth":"640","minHeight":"200","maxHeight":"480"}}}
main.js:93 GetUserMedia succeeded
main.js:262 Not connected yet
main.js:141 localPeerConnection creating offer
main.js:168 localPeerConnection offering
8main.js:149 Candidate localPeerConnection
main.js:172 remotePeerConnection answering
main.js:164 remotePeerConnection got stream
common.js:8 5.820: AddIceCandidate success.
common.js:8 5.821: AddIceCandidate success.
common.js:8 5.822: AddIceCandidate success.
2common.js:8 5.837: AddIceCandidate success.
common.js:8 5.838: AddIceCandidate success.
common.js:8 5.839: AddIceCandidate success.
common.js:8 5.840: AddIceCandidate success.
2main.js:149 Candidate localPeerConnection
2main.js:156 Candidate remotePeerConnection
common.js:8 5.944: AddIceCandidate success.
common.js:8 6.044: AddIceCandidate success.
common.js:8 6.045: AddIceCandidate success.
common.js:8 6.047: AddIceCandidate success.
main.js:156 Candidate remotePeerConnection
main.js:149 Candidate localPeerConnection
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
main.js:127 getUserMedia constraints Object {audio: true, video: Object}
common.js:8 42.067: Ending call
2main.js:262 Not connected yet
adapter.js:32 chrome: {"audio":true,"video":{"mandatory":{"minWidth":"800","maxWidth":"1000","minHeight":"690","maxHeight":"800"}}}
main.js:93 GetUserMedia succeeded
5main.js:262 Not connected yet
main.js:141 localPeerConnection creating offer
main.js:168 localPeerConnection offering
main.js:172 remotePeerConnection answering
8main.js:149 Candidate localPeerConnection
main.js:164 remotePeerConnection got stream
common.js:8 49.455: AddIceCandidate success.
common.js:8 49.456: AddIceCandidate success.
3common.js:8 49.457: AddIceCandidate success.
common.js:8 49.458: AddIceCandidate success.
2common.js:8 49.459: AddIceCandidate success.
3main.js:156 Candidate remotePeerConnection
2main.js:149 Candidate localPeerConnection
common.js:8 49.498: AddIceCandidate success.
common.js:8 49.499: AddIceCandidate success.
2common.js:8 49.501: AddIceCandidate success.
main.js:149 Candidate localPeerConnection"

**Comment by KaptenJansson** [comment]
Something weird is happening here, why are there so many main.js:127 getUserMedia constraints Object {audio: true, video: Object} printouts in the log? I do not think that's relevant to this issue however.

This looks like a hardware/device specific issue, not related to this demo page. Is the video corrupt using other constraints? Also, what happens if you do the same using an external camera, like the Logitech 520/C910/C920?

**Comment by cvranjani** [comment]
@KaptenJansson 

main.js:127 getUserMedia constraints Object {audio: true, video: Object} shows so many entries in the javascript logs as I was adjusting the constraints using up and down arrow keys in between until I got the desired {"minWidth":"800","maxWidth":"1000","minHeight":"690","maxHeight":"800"}}}

I  have seen this issue on two devices Jerry and Big with M49 version but I am unable to reproduce this issue on the same devices now.

Closing this bug. Thanks.
